### PR TITLE
Add libvirt dbgsym in debian

### PR DESCRIPTION
**Pull request by KKoukiou**
* [x] image-refresh debian-stable
 * [x] image-refresh debian-testing

### CODE REVIEW

**Comment by cockpituous** [comment]
image-refresh in progress on 2-cockpit-9.
Log: https://logs.cockpit-project.org/logs/image-refresh-1350-20201030-142525/

**Comment by cockpituous** [comment]
image-refresh debian-stable done: https://github.com/cockpituous/bots/commits/image-refresh-debian-stable-20201030-143737

**Comment by cockpituous** [comment]
image-refresh in progress on 2-cockpit-9.
Log: https://logs.cockpit-project.org/logs/image-refresh-1350-20201030-143838/

**Comment by cockpituous** [comment]
image-refresh debian-testing done: https://github.com/cockpituous/bots/commits/image-refresh-debian-testing-20201030-145757

File: images/scripts/debian.setup [hunk 1]
```diff
@@ -86,6 +92,10 @@ case "$RELEASE" in
         IPA_CLIENT_PACKAGES="${IPA_CLIENT_PACKAGES/freeipa-client /}"
 esac
 
+cat <<EOF | sudo tee /etc/apt/sources.list.d/dbgsym.list
+deb http://debug.mirrors.debian.org/debian-debug/ ${RELEASE}-debug main
+EOF
```
Comment (martinpitt) [inline images/scripts/debian.setup:97]: This is debian specific, and doesn't apply to the Ubuntu images. So they need to be guarded. Maybe easiest to move them into the `case` above?

**Review by martinpitt — CHANGES_REQUESTED** [review]
Do we have more crashes than just the [old D-Bus one on shutdown](https://github.com/cockpit-project/bots/issues/798)? But either way, good call on making backtraces more useful by default!

File: images/scripts/debian.setup (hunk 1 above)
Comment (KKoukiou) [reply images/scripts/debian.setup:97]: Oh good catch, I will. 
Yes, we have some crashes which are not naughted.

**Comment by cockpituous** [comment]
image-refresh in progress on centosci-tasks-tgpkq.
Log: https://logs-https-frontdoor.apps.ocp.ci.centos.org/logs/image-refresh-1350-20201030-154646/

**Comment by cockpituous** [comment]
image-refresh debian-stable done: https://github.com/cockpituous/bots/commits/image-refresh-debian-stable-20201030-155858

**Comment by cockpituous** [comment]
image-refresh in progress on 2-ci-srv-01.
Log: https://logs.cockpit-project.org/logs/image-refresh-1350-20201030-160404/

**Comment by cockpituous** [comment]
image-refresh debian-testing done: https://github.com/cockpituous/bots/commits/image-refresh-debian-testing-20201030-161717

File: images/scripts/debian.setup (hunk 1 above)
Comment (martinpitt) [reply images/scripts/debian.setup]: Thanks! That works. Not approving yet, as you still need to push the rebuilds and test them, but :+1:

**Comment by marusak** [comment]
@KKoukiou you forgot to trigger `debian-testing@cockpit-project/cockpit-podman`.
I actually advice using `./tests-trigger --repo cockpit-project/bots 1350 image:debian-testing` syntax when not sure what all tests need to be triggered for image refresh. (the `image:<foo>` syntax)

And I forgot to update my bots so by mistake I triggered also `host` context :) Please ignore that one

**Comment by KKoukiou** [comment]
Hmm I am not sure why but coredumpctl does not want to register the coredumps.
I see:
https://logs.cockpit-project.org/logs/pull-1350-20201102-112645-6ac2bd1c-debian-stable-cockpit-project-cockpit/TestMachines-testInlineConsoleWithUrlRoot-debian-stable-127.0.0.2-2301-FAIL.log.gz

**Comment by marusak** [comment]
> Hmm I am not sure why but coredumpctl does not want to register the coredumps.

`ulimit -c`?

**Review by martinpitt — DISMISSED** [review]
Thanks!

**Comment by martinpitt** [comment]
That new commit looks good. Triggering tests, so that we get some results.

**Comment by martinpitt** [comment]
@KKoukiou Nice, a [somewhat useful backtrace](https://logs.cockpit-project.org/logs/pull-1350-20201103-090725-fe2ff21e-debian-stable-cockpit-project-cockpit/log.html#83)!

**Review by martinpitt — APPROVED** [review]

**Comment by KKoukiou** [comment]
Hmm I think I need some more of the debug info packages.

The stacktrace is not very helpful

Stack trace of thread 1876:
#0  0x00007f3c8552e6c0 __GI___pthread_mutex_lock (libpthread.so.0)
#1  0x00007f3c7c1f54de virQEMUDriverGetConfig (libvirt_driver_qemu.so)
#2  0x00007f3c7c23a29c n/a (libvirt_driver_qemu.so)
#3  0x00007f3c8605245f virStateStop (libvirt.so.0)
#4  0x000055d8fda6d3fd n/a (libvirtd)
#5  0x00007f3c85eba382 virThreadHelper (libvirt.so.0)
#6  0x00007f3c8552bfa3 start_thread (libpthread.so.0)
#7  0x00007f3c854504cf __clone (libc.so.6)
Stack trace of thread 1223:
#0  0x00007f3c8553200c futex_wait_cancelable (libpthread.so.0)
#1  0x00007f3c85eba5b6 virCondWait (libvirt.so.0)
#2  0x00007f3c85ebb0b3 virThreadPoolWorker (libvirt.so.0)
#3  0x00007f3c85eba358 virThreadHelper (libvirt.so.0)
#4  0x00007f3c8552bfa3 start_thread (libpthread.so.0)
#5  0x00007f3c854504cf __clone (libc.so.6)
Stack trace of thread 1224:
#0  0x00007f3c8553200c futex_wait_cancelable (libpthread.so.0)
#1  0x00007f3c85eba5b6 virCondWait (libvirt.so.0)
#2  0x00007f3c85ebb0b3 virThreadPoolWorker (libvirt.so.0)
#3  0x00007f3c85eba358 virThreadHelper (libvirt.so.0)
#4  0x00007f3c8552bfa3 start_thread (libpthread.so.0)
#5  0x00007f3c854504cf __clone (libc.so.6)
Stack trace of thread 1225:
#0  0x00007f3c8553200c futex_wait_cancelable (libpthread.so.0)
#1  0x00007f3c85eba5b6 virCondWait (libvirt.so.0)
#2  0x00007f3c85ebb0b3 virThreadPoolWorker (libvirt.so.0)
#3  0x00007f3c85eba358 virThreadHelper (libvirt.so.0)
#4  0x00007f3c8552bfa3 start_thread (libpthread.so.0)
#5  0x00007f3c854504cf __clone (libc.so.6)
Stack trace of thread 1226:
#0  0x00007f3c8553200c futex_wait_cancelable (libpthread.so.0)
#1  0x00007f3c85eba5b6 virCondWait (libvirt.so.0)
#2  0x00007f3c85ebb0b3 virThreadPoolWorker (libvirt.so.0)
#3  0x00007f3c85eba358 virThreadHelper (libvirt.so.0)
#4  0x00007f3c8552bfa3 start_thread (libpthread.so.0)
#5  0x00007f3c854504cf __clone (libc.so.6)
Stack trace of thread 1227:
#0  0x00007f3c8553200c futex_wait_cancelable (libpthread.so.0)
#1  0x00007f3c85eba5b6 virCondWait (libvirt.so.0)
#2  0x00007f3c85ebb0b3 virThreadPoolWorker (libvirt.so.0)
#3  0x00007f3c85eba358 virThreadHelper (libvirt.so.0)
#4  0x00007f3c8552bfa3 start_thread (libpthread.so.0)
#5  0x00007f3c854504cf __clone (libc.so.6)
Stack trace of thread 1228:
#0  0x00007f3c8553200c futex_wait_cancelable (libpthread.so.0)
#1  0x00007f3c85eba5b6 virCondWait (libvirt.so.0)
#2  0x00007f3c85ebb064 virThreadPoolWorker (libvirt.so.0)
#3  0x00007f3c85eba358 virThreadHelper (libvirt.so.0)
#4  0x00007f3c8552bfa3 start_thread (libpthread.so.0)
#5  0x00007f3c854504cf __clone (libc.so.6)
Stack trace of thread 1229:
#0  0x00007f3c8553200c futex_wait_cancelable (libpthread.so.0)
#1  0x00007f3c85eba5b6 virCondWait (libvirt.so.0)
#2  0x00007f3c85ebb064 virThreadPoolWorker (libvirt.so.0)
#3  0x00007f3c85eba358 virThreadHelper (libvirt.so.0)
#4  0x00007f3c8552bfa3 start_thread (libpthread.so.0)
#5  0x00007f3c854504cf __clone (libc.so.6)
Stack trace of thread 1230:
#0  0x00007f3c8553200c futex_wait_cancelable (libpthread.so.0)
#1  0x00007f3c85eba5b6 virCondWait (libvirt.so.0)
#2  0x00007f3c85ebb064 virThreadPoolWorker (libvirt.so.0)
#3  0x00007f3c85eba358 virThreadHelper (libvirt.so.0)
#4  0x00007f3c8552bfa3 start_thread (libpthread.so.0)
#5  0x00007f3c854504cf __clone (libc.so.6)
Stack trace of thread 1231:
#0  0x00007f3c8553200c futex_wait_cancelable (libpthread.so.0)
#1  0x00007f3c85eba5b6 virCondWait (libvirt.so.0)
#2  0x00007f3c85ebb064 virThreadPoolWorker (libvirt.so.0)
#3  0x00007f3c85eba358 virThreadHelper (libvirt.so.0)
#4  0x00007f3c8552bfa3 start_thread (libpthread.so.0)
#5  0x00007f3c854504cf __clone (libc.so.6)
Stack trace of thread 1232:
#0  0x00007f3c8553200c futex_wait_cancelable (libpthread.so.0)
#1  0x00007f3c85eba5b6 virCondWait (libvirt.so.0)
#2  0x00007f3c85ebb064 virThreadPoolWorker (libvirt.so.0)
#3  0x00007f3c85eba358 virThreadHelper (libvirt.so.0)
#4  0x00007f3c8552bfa3 start_thread (libpthread.so.0)
#5  0x00007f3c854504cf __clone (libc.so.6)
Stack trace of thread 1235:
#0  0x00007f3c8553200c futex_wait_cancelable (libpthread.so.0)
#1  0x00007f3c85eba5b6 virCondWait (libvirt.so.0)
#2  0x00007f3c85ebb0b3 virThreadPoolWorker (libvirt.so.0)
#3  0x00007f3c85eba358 virThreadHelper (libvirt.so.0)
#4  0x00007f3c8552bfa3 start_thread (libpthread.so.0)
#5  0x00007f3c854504cf __clone (libc.so.6)
Stack trace of thread 1236:
#0  0x00007f3c8553200c futex_wait_cancelable (libpthread.so.0)
#1  0x00007f3c85eba5b6 virCondWait (libvirt.so.0)
#2  0x00007f3c85ebb0b3 virThreadPoolWorker (libvirt.so.0)
#3  0x00007f3c85eba358 virThreadHelper (libvirt.so.0)
#4  0x00007f3c8552bfa3 start_thread (libpthread.so.0)
#5  0x00007f3c854504cf __clone (libc.so.6)
Stack trace of thread 1237:
#0  0x00007f3c8553200c futex_wait_cancelable (libpthread.so.0)
#1  0x00007f3c85eba5b6 virCondWait (libvirt.so.0)
#2  0x00007f3c85ebb0b3 virThreadPoolWorker (libvirt.so.0)
#3  0x00007f3c85eba358 virThreadHelper (libvirt.so.0)
#4  0x00007f3c8552bfa3 start_thread (libpthread.so.0)
#5  0x00007f3c854504cf __clone (libc.so.6)
Stack trace of thread 1238:
#0  0x00007f3c8553200c futex_wait_cancelable (libpthread.so.0)
#1  0x00007f3c85eba5b6 virCondWait (libvirt.so.0)
#2  0x00007f3c85ebb0b3 virThreadPoolWorker (libvirt.so.0)
#3  0x00007f3c85eba358 virThreadHelper (libvirt.so.0)
#4  0x00007f3c8552bfa3 start_thread (libpthread.so.0)
#5  0x00007f3c854504cf __clone (libc.so.6)
Stack trace of thread 1239:
#0  0x00007f3c8553200c futex_wait_cancelable (libpthread.so.0)
#1  0x00007f3c85eba5b6 virCondWait (libvirt.so.0)
#2  0x00007f3c85ebb0b3 virThreadPoolWorker (libvirt.so.0)
#3  0x00007f3c85eba358 virThreadHelper (libvirt.so.0)
#4  0x00007f3c8552bfa3 start_thread (libpthread.so.0)
#5  0x00007f3c854504cf __clone (libc.so.6)
Stack trace of thread 1218:
#0  0x00007f3c853db9c1 __GI___libc_free (libc.so.6)
#1  0x00007f3c85c696b0 xmlHashFree (libxml2.so.2)
#2  0x00007f3c7cadc1b9 xsltFreeTemplateHashes (libxslt.so.1)
#3  0x00007f3c7cad0fd8 xsltFreeStylesheet (libxslt.so.1)
#4  0x00007f3c7cb7ad18 n/a (libnetcf.so.1)
#5  0x00007f3c7cb757bf ncf_close (libnetcf.so.1)
#6  0x00007f3c85e9262b virObjectUnref (libvirt.so.0)
#7  0x00007f3c7cd9dff5 n/a (libvirt_driver_interface.so)
#8  0x00007f3c86052380 virStateCleanup (libvirt.so.0)
#9  0x000055d8fda6c2d3 n/a (libvirtd)
#10 0x00007f3c8537b09b __libc_start_main (libc.so.6)
#11 0x000055d8fda6d11a n/a (libvirtd)

**Comment by KKoukiou** [comment]
I will merge, and figure out in another PR which libvirt dbgsym packages are missing.

**Comment by martinpitt** [comment]
@KKoukiou : I'd say don't bother -- we [get a core dump now](https://logs.cockpit-project.org/logs/pull-1350-20201103-090725-fe2ff21e-debian-stable-cockpit-project-cockpit/TestMachines-testAddDiskNFS-debian-stable-127.0.0.2-2301-FAIL.core/), so you can always boot the image, install the other dbgsyms, and re-do the trace in gdb. That avoids balooning the images too much with large dbgsym packages.

**Pull request merged**